src/hotspot/share/runtime/sharedRuntime.cpp

```diff
@@ -2954,7 +2961,7 @@ VMRegPair *SharedRuntime::find_callee_arguments(Symbol* sig, bool has_receiver,
   // This method is returning a data structure allocating as a
   // ResourceObject, so do not put any ResourceMarks in here.
 
-  BasicType *sig_bt = NEW_RESOURCE_ARRAY(BasicType, 256);
+  BasicType sig_bt[256];
```

Comment:
Was this line changed by mistake? The rest of the PR looks good to me.

Reply:
It was intentional, but it's also a bit of a scope creep: this method is about creating and returning the VMRegPair array, but it also resource-allocs this sig_bt array and keeps it alive for however long the regs array is kept alive, which would be a memory leak if not for the fact that it's mostly used in a transient way.

Reply:
The think the original code was written to avoid allocating large arrays on the stack.

Reply:
Ok, I reverted this change. It's not part of anything I know sees much use anyhow.